oa4mp/serve.go

```diff
@@ -306,6 +306,27 @@ func ConfigureOA4MP() (launcher daemon.Launcher, err error) {
 		return
 	}
 
+	// Ensure the OA4MP storage directory exists and has correct permissions and ownership
+	OA4MPStoragePath := "/opt/scitokens-server/var/storage"
+
+	if err := os.Chmod(OA4MPStoragePath, 0700); err != nil {
+		if os.IsNotExist(err) {
+			log.Debugln("OA4MP storage directory does not exist. Creating", OA4MPStoragePath)
+			if err = os.MkdirAll(OA4MPStoragePath, 0755); err != nil {
+				log.Fatal("Failed to create directory", OA4MPStoragePath, ":", err)
+			}
+			if err = os.Chmod(OA4MPStoragePath, 0700); err != nil {
+				log.Errorln("Failed to change permissions of", OA4MPStoragePath, "after creating it:", err)
```

Comment:
Same here. Instead of logging the error and continuing, do something like:
```
if err = os.Chmod(OA4MPStoragePath, 0700); err != nil {
    err = errors.Wrapf(err, "failed to change permissions of %s after creating it")
    return
}
```